pkg/services/internal/bitails/utils.go

```diff
@@ -59,6 +60,26 @@ func ConvertHeader(raw []byte, height uint32) (*wdk.ChainBlockHeader, error) {
 	}, nil
 }
 
+func validateScriptHash(scriptHash string) error {
+	switch {
+	case scriptHash == "":
+		return fmt.Errorf("scripthash cannot be empty")
+	case len(scriptHash)%2 != 0:
+		return fmt.Errorf("invalid scripthash length: must be even")
+	case len(scriptHash) != 64:
+		return fmt.Errorf("invalid scripthash length: must be 64 characters for a SHA256 hash")
+	case !isHex(scriptHash):
+		return fmt.Errorf("invalid scripthash format: must be hex encoded")
+	default:
+		return nil
+	}
```

Comment:
The general-purpose HexString validation is already implemenetd in the "primitives" package. Consider creating specialized type in the primitives or use following implementation. 

```suggestion
	if scriptHash == "" {
		return fmt.Errorf("scripthash cannot be empty")
	}

	if len(scriptHash) != 64 {
		return fmt.Errorf("invalid scripthash length: must be 64 characters for a SHA256 hash")
	}

	if err := primitives.HexString(scriptHash).Validate(); err != nil {
		return fmt.Errorf("invalid scripthash format: %w", err)
	}

	return nil
```